http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/https/CipherSuitesTest.java

```diff
@@ -0,0 +1,184 @@
+
+        TransportsConfiguration transportsConfiguration = TestUtil
+                .getConfiguration("/simple-test-config" + File.separator + "netty-transports.yml");
+        Set<SenderConfiguration> senderConfig = transportsConfiguration.getSenderConfigurations();
+        senderConfig.forEach(config -> {
+            config.setKeyStoreFile(keyStoreFile);
+            config.setKeyStorePassword(password);
+            config.setParameters(clientParams);
+        });
+
+        HttpWsConnectorFactory factory = new HttpWsConnectorFactoryImpl();
+        ListenerConfiguration listenerConfiguration = ListenerConfiguration.getDefault();
+        listenerConfiguration.setPort(serverPort);
+        listenerConfiguration.setVerifyClient(verifyClient);
+        listenerConfiguration.setTrustStoreFile(trustStoreFile);
+        listenerConfiguration.setKeyStoreFile(keyStoreFile);
+        listenerConfiguration.setTrustStorePass(password);
+        listenerConfiguration.setKeyStorePass(password);
+        listenerConfiguration.setScheme(scheme);
+        listenerConfiguration.setParameters(serverParams);
+
+        ServerConnector serverConnector = factory
+                .createServerConnector(ServerBootstrapConfiguration.getInstance(), listenerConfiguration);
+        ServerConnectorFuture future = serverConnector.start();
+        future.setHttpConnectorListener(new EchoMessageListener());
+        future.sync();
+
+        httpClientConnector = factory
+                .createHttpClientConnector(HTTPConnectorUtil.getTransportProperties(transportsConfiguration),
+                        HTTPConnectorUtil.getSenderConfiguration(transportsConfiguration, Constants.HTTPS_SCHEME));
+
+        testCiphersuites(hasException, serverPort);
+        serverConnector.stop();
+        httpClientConnector.close();
+    }
+
+    public void testCiphersuites(boolean hasException, int serverPort) {
+        try {
+            ByteBuffer byteBuffer = ByteBuffer.wrap(testValue.getBytes(Charset.forName("UTF-8")));
+            HTTPCarbonMessage msg = new HTTPCarbonMessage(
+                    new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET, ""));
+            msg.setProperty("PORT", serverPort);
+            msg.setProperty("PROTOCOL", scheme);
+            msg.setProperty("HOST", TestUtil.TEST_HOST);
+            msg.setProperty("HTTP_METHOD", "GET");
+            msg.addMessageBody(byteBuffer);
+            msg.setEndOfMsgAdded(true);
+
+            CountDownLatch latch = new CountDownLatch(1);
+            SSLConnectorListener listener = new SSLConnectorListener(latch);
+            HttpResponseFuture responseFuture = httpClientConnector.send(msg);
+            responseFuture.setHttpConnectorListener(listener);
+            latch.await(5, TimeUnit.SECONDS);
+            HTTPCarbonMessage response = listener.getHttpResponseMessage();
+
+            if (hasException) {
+                assertNotNull(listener.getThrowables());
```

Comment:
Shall we add assert messages?

Reply:
At this time I need to know whether there is an exception throwing or not. If there is, the thrown exception will be tested (whether it is an ssl exception) in the latter part of the code.